lymph/midline.py

```diff
@@ -520,20 +340,25 @@ def _evolve_midext(
 
         :meta public:
         """
+        if start_midexstate is None:
+            start_midexstate = np.zeros(
+            shape=(self.diag_time_dists.max_t + 1, 2),
+            dtype=float
+            )
+            start_midexstate[0,0] = 1.
+        
+        self.midext_prob = new_params
 
-        # compute involvement at first time-step
-        new_midexstate = start_midexstate @ self.midextransition_matrix
-
-        return new_midexstate
+        midextransition_matrix = np.zeros(shape=(2**1, 2**1))
+        midextransition_matrix[0,0] = 1 - self.midext_prob
+        midextransition_matrix[0,1] = self.midext_prob
+        midextransition_matrix[1,0] = 0
+        midextransition_matrix[1,1] = 1
 
-    """
-    def state_probs_midext(self):
-        max_t = self.diag_time_dists.max_t
-        self.ext.state_probs_midext = {}
-        self.noext.state_probs_midext = {}
-        self.noext.state_probs_midext = self.noext._evolve_midext(t_last=max_t)
-        self.ext.state_probs_midext = self.ext.
-    """
+        # compute involvement at first time-step
+        for i in range(len(start_midexstate)-1):
+            start_midexstate[i+1,:] = start_midexstate[i,:] @ midextransition_matrix
```

Comment:
This computes the midline extension state for all time-steps, right? Not just for the first one, as the comment suggests.

Reply:
Of course, you're right.